draft-dss-star.md

```diff
@@ -101,200 +121,640 @@ In this document we describe one such system, namely Distributed Secret Sharing
+Upon receipt, the client computes `evaluated_element = DeserializeElement(randomness_response)`.
+If this fails, the client aborts the protocol. Otherwise, the client
+finalizes the OPRF protocol to compute the output `rand` as follows:
 
-Finally, note that under collusion, the STAR security model requires the adversary to launch an offline dictionary attack against client measurements. In Prio and Poplar, security is immediately lost when collusion occurs.
+```
+rand = client_context.Finalize(msg, blind, evaluated_element)
+```
 
-## Support for auxiliary data
+## Reporting Phase {#client-message}
+
+In the reporting phase, the client uses its measurement `msg` with auxiliary data `aux`
+and its derived randomness `rand` to produce a report for the Aggregation Server.
+
+### Reporting Configuration
+
+The reporting phase requires the Aggregation Server to be configured with a URI for
+accepting reports. As an example, the Aggregation Server URI might be https://aggregator.example.
+The Aggregation Server is both an Oblivious HTTP Target and Oblivious Gateway Resource.
 
-It should be noted that clients can send auxiliary data ({{auxiliary-data}}) that is revealed only when the aggregation including their measurement succeeds (i.e. K-1 other clients send the same value). Such data is supported by neither Prio, nor Poplar.
+Clients are also configured with an Anonymizing Proxy that clients can use to send
+proxy reports to the Aggregation Server. The exact type of proxy is not specified here.
+See {{proxy-options}} for more details.
+
+### Reporting Protocol
+
+This reporting protocol works as follows. First, the client stretches `rand` into three values
+`key_seed`, `share_coins`, and `tag`, and additionally derives an KCAEAD key and nonce
+from `key_seed`.
+
+```
+// Randomness derivation
+rand_prk = Extract(nil, rand)
+key_seed = Expand(rand_prk, "key_seed", 16)
+share_coins = Expand(rand_prk, "share_coins", 16)
+tag = Expand(rand_prk, "tag", 16)
+
+// Symmetric encryption key derivation
+key_prk = Extract(nil, key_seed)
+key = Expand(key_prk, "key", Nk)
+nonce = Expand(key_prk, "nonce", Nn)
+```
+
+The client then generates a secret share of `key_seed` using `share_coins` as randomness as follows:
+
+```
+rand_share = Share(REPORT_THRESHOLD, TBD, key_seed, share_coins, nil)
+```
+
+[[OPEN ISSUE: what should N be for the TSS scheme?]]
+
+The client then encrypts `msg` and `aux` using the KCAEAD key and nonce as follows:
+
+```
+report_data = msg || aux
+encrypted_report = Seal(key, nonce, nil, report_data)
+```
+
+Finally, the client constructs a report consisting of `encrypted_report`, `rand_share`,
+and `tag`, and sends this to the Anonymizing Server in the subsequent epoch, i.e., after
+the Randomness Server has rotated its OPRF key.
+
+[[OPEN ISSUE: how does the client know when this rotation happens?]]
```

Comment:
This is one of the advantages of using a VOPRF, they can check that the correct key is being used for the time period.

Reply:
That would be a good thing to add.

Reply:
Let's do so in a followup PR.